source/clear-linux/tutorials/kata.rst

```diff
@@ -0,0 +1,81 @@
+  .. _kata:
+
+Install Kata Containers
+#######################
+
+This tutorial describes how to install, configure and run Kata Containers on
+|CLOSIA|. Kata Containers is an open source project and community working to
+build a standard implementation of lightweight Virtual Machines (VMs) that feel
+and perform like containers, but provide the workload isolation and security
+advantages of VMs. Kata Containers is available in Clear Linux versions above 22860.
+
+Prerequisites
+*************
+
+This tutorial assumes you have installed |CL| on your host system.
+For detailed instructions on installing |CL| on a bare metal system, visit
+the :ref:`bare metal installation tutorial<bare-metal-install>`.
+
+Before you install any new packages, update |CL| with the following command:
+
+.. code-block:: bash
+
+   sudo swupd update
+
+Install Kata Containers
+***********************
+
+Kata Containers is included in the :file:`containers-virt` bundle. To install the
+framework, enter:
+
+.. code-block:: bash
+
+   sudo swupd bundle-add containers-virt
+
+Configure Docker to use Kata Containers by default
+
+.. code-block:: bash
+
+   sudo mkdir -p /etc/systemd/system/docker.service.d/
+   cat <<EOF | sudo tee /etc/systemd/system/docker.service.d/kata-containers.conf
+   [Service]
+   ExecStart=
+   ExecStart=/usr/bin/dockerd -D --add-runtime kata-runtime=/usr/bin/kata-runtime --default-runtime=kata-runtime
+   EOF
+   
+Restart the Docker\* and Kata Containers systemd services
+
+.. code-block:: bash
+
+   sudo systemctl daemon-reload
+   sudo systemctl restart docker
+
+Run Kata Containers
+*******************
+
+.. code-block:: bash
+
+   sudo docker run -ti busybox sh
```

Comment:
I got the following error:
Unable to find image 'busybox:latest' locally
docker: Error response from daemon: Get https://registry-1.docker.io/v2/: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers).
See 'docker run --help'.

I tried using sudo and then sudo -E and both resulted in this error message

Reply:
Hi @tpleavitt, thanks for your review. At this point the "sudo swupd bundle-add containers-virt" command does not install the required kata-containers package (because the updated bundle hasn't been made publicly available), so the described configuration for docker to use kata-containers is not working. Right now, if you want to validate the tutorial you must manually build and install the following packages in Clear Linux:
kata-image, kata-proxy, kata-qemu-lite, kata-runtime, kata-shim and linux-kata.
They're all at the koji repository.
If this is problematic for you, we can pause the revision until the updated bundle is out.

Reply:
Same situation (in my previous comment) applies to pull request #154

Reply:
OK, I'm on 22870 and it said it would be available in 22860 above.  When will it be released?  I'll wait for that.

Reply:
@tpleavitt I can try this tutorial too. Is that OK with you Tom? If not, no problem. Either way, I should wait until the command works: `sudo swupd bundle-add containers-virt`

Reply:
@mvincerx definitely!  The bundle already exists and loaded fine but it is missing components that will be added soon.  We'll need to change the reference build above once it is available in whichever version it gets released in.  It wasn't added to 22880 either.

Reply:
Correct. Some last-minute tweaks are being performed to the missing packages. I will let you guys know when the updated bundle is available. I will also update the version number.

Reply:
@joselamego I'm still seeing this problem with 23550. Otherwise LGTM.

Reply:
@intelkevinputnam , Daniela sent this to me an an email a few weeks ago, should solve the proxy issue:
Despite having proxy variables set on your env, you must use a config file setting those for Docker:
$ mkdir -p /etc/systemd/system/docker.service.d
$ cat <<EOT | tee /etc/systemd/system/docker.service.d/proxy.conf
[Service]
Environment=HTTP_PROXY=$http_proxy
Environment=HTTPS_PROXY=$https_proxy
EOT
$ systemctl daemon-reload
$ systemctl restart docker
That should solve your issue.